### PR TITLE
Connector for DB provide default port

### PR DESCRIPTION
**Pull request by JimMoen**

### CODE REVIEW

File: apps/emqx_connector/src/emqx_connector_redis.erl [hunk 1]
```diff
@@ -52,6 +37,21 @@
 
 -export([cmd/3]).
 
+
+%% redis server(s) parse
+-type server_raw() :: tuple().
+-type servers_raw() :: [server_raw()].
```
Comment (zmstone) [inline apps/emqx_connector/src/emqx_connector_redis.erl:43]: try to use primitive types and add use converter + validator for the field.

https://github.com/emqx/hocon/blob/0cf2a9e1173aad39bdbd35445f74a755e4707f20/src/hocon_schema.erl#L80-L81

primitive types such as `string()` is more readable in the generated document.
the parsing rules can be explained in `desc` schema field.

File: apps/emqx_authn/src/simple_authn/emqx_authn_pgsql.erl [hunk 2]
```diff
@@ -54,9 +54,9 @@ fields(?CONF_NS) ->
     , {backend, emqx_authn_schema:backend(postgresql)}
     , {password_hash_algorithm, fun emqx_authn_password_hashing:type_ro/1}
     , {query,                   fun query/1}
-    ] ++ emqx_authn_schema:common_fields()
-    ++ emqx_connector_schema_lib:relational_db_fields()
-    ++ emqx_connector_schema_lib:ssl_fields().
+    ] ++
+    emqx_authn_schema:common_fields() ++
```
Comment (zhongwencool) [inline apps/emqx_authn/src/simple_authn/emqx_authn_pgsql.erl:58]: I find `emqx_connector_schema_lib:relational_db_fields() ++ emqx_connector_schema_lib:ssl_fields()`  always appear at the same place. maybe we should export a function call `emqx_connector_schema_lib:relational_db_with_ssl_fields()`.
```erlang

```

Comment (JimMoen) [reply apps/emqx_authn/src/simple_authn/emqx_authn_pgsql.erl:58]: They are two different reusable part fields for DB connector.
Maybe shouldn't in single function.

File: apps/emqx_authn/test/emqx_authn_mongo_tls_SUITE.erl [hunk 3]
```diff
@@ -169,15 +169,15 @@ raw_mongo_auth_config(SpecificSSLOpts) ->
...
       ssl => maps:merge(SSLOpts, SpecificSSLOpts)
      }.
 
 mongo_server() ->
     iolist_to_binary(
       io_lib:format(
-        "~s:~b",
-        [?MONGO_HOST, ?MONGO_PORT])).
+        "~s",
+        [?MONGO_HOST])).
```
Comment (terry-xiaoyu) [inline apps/emqx_authn/test/emqx_authn_mongo_tls_SUITE.erl:180]: I don't like such code style as it puts only one parameter in one line, which increase total line of the source file.

**Review by terry-xiaoyu — APPROVED** [review]

File: apps/emqx_authn/test/emqx_authn_mongo_tls_SUITE.erl (hunk 3 above)
Comment (JimMoen) [reply apps/emqx_authn/test/emqx_authn_mongo_tls_SUITE.erl:180]: Didn't think so much. 
Will be changed to a single line in conflict resolution.
(Caused by _"`on_jsonify/1` removed"_ merged)

**Review by zhongwencool — APPROVED** [review]

File: apps/emqx_connector/include/emqx_connector.hrl [hunk 4]
```diff
@@ -1,4 +1,39 @@
...
 -define(MAX(MAXV), ?VALID:max(number, MAXV)).
 -define(MIN(MINV), ?VALID:min(number, MINV)).
+
+-define(MYSQL_DEFAULT_PORT, 3306).
+-define(MONGO_DEFAULT_PORT, 27017).
+-define(REDIS_DEFAULT_PORT, 6379).
+-define(PGSQL_DEFAULT_PORT, 5432).
+
+-define(SERVERS_DESC, """
+A Node list to for Cluster to connect to.<br>
```
Comment (HJianBo) [inline apps/emqx_connector/include/emqx_connector.hrl:28]: plz re-phrase it

**Review by HJianBo — CHANGES_REQUESTED** [review]

File: apps/emqx_connector/src/emqx_connector_mongo.erl [hunk 5]
```diff
@@ -315,47 +318,52 @@ may_parse_srv_and_txt_records_(#{mongo_type := Type,
...
+                      [] ->
+                          error(service_not_found);
+                      Services ->
+                          [ [server_to_bin({Host, Port}) || {_, _, Port, Host} <- Services]
+                          | AccIn]
+                  end
+          end,
+    Res = lists:foldl(Fun, [], Servers),
+    case Type of
+        single -> lists:nth(1, Res);
```
Comment (HJianBo) [inline apps/emqx_connector/src/emqx_connector_mongo.erl:342]: Do we need a warning log? if the Res length > 1

Comment (JimMoen) [reply apps/emqx_connector/src/emqx_connector_mongo.erl:342]: Log with srv record `priority` and `weight` parse in next PR.

**Review by HJianBo — APPROVED** [review]

**Review by HJianBo — APPROVED** [review]

**Pull request merged**